### PR TITLE
fix(filter): Fix two bugs.

### PR DESCRIPTION
**Pull request by irevoire**
- Stop lowercasing the field when looking in the field id map
- When a field id does not exist it means there is currently zero
  documents containing this field thus we return an empty RoaringBitmap
  instead of throwing an internal error

Will fix https://github.com/meilisearch/MeiliSearch/issues/2082 once meilisearch is released

### CODE REVIEW

File: milli/src/search/facet/filter.rs [hunk 1]
```diff
@@ -342,12 +340,12 @@ impl<'a> Filter<'a> {
         match &self.condition {
             FilterCondition::Condition { fid, op } => {
                 let filterable_fields = index.filterable_fields(rtxn)?;
-                if filterable_fields.contains(&fid.to_lowercase()) {
+                if filterable_fields.contains(fid.value()) {
```
Comment (Kerollmops) [inline milli/src/search/facet/filter.rs:343]: Don't you implement `Deref` on that type? Do you need to call `value()` here?

File: milli/src/search/facet/filter.rs [hunk 2]
```diff
@@ -58,7 +57,6 @@ impl<'a> Display for FilterError<'a> {
             Self::BadGeo(keyword) => write!(f, "`{}` is a reserved keyword and thus can't be used as a filter expression. Use the _geoRadius(latitude, longitude, distance) built-in rule to filter on _geo field coordinates.", keyword),
             Self::BadGeoLat(lat) => write!(f, "Bad latitude `{}`. Latitude must be contained between -90 and 90 degrees. ", lat),
             Self::BadGeoLng(lng) => write!(f, "Bad longitude `{}`. Longitude must be contained between -180 and 180 degrees. ", lng),
-            Self::InternalError => write!(f, "Internal error while executing this filter."),
```
Comment (Kerollmops) [inline milli/src/search/facet/filter.rs:61]: Ho! I'm glad we remove this variant! 😃

File: milli/src/search/facet/filter.rs [hunk 3]
```diff
@@ -342,12 +340,12 @@ impl<'a> Filter<'a> {
...
                 let filterable_fields = index.filterable_fields(rtxn)?;
-                if filterable_fields.contains(&fid.to_lowercase()) {
+                if filterable_fields.contains(fid.value()) {
                     let field_ids_map = index.fields_ids_map(rtxn)?;
-                    if let Some(fid) = field_ids_map.id(&fid) {
+                    if let Some(fid) = field_ids_map.id(fid.value()) {
                         Self::evaluate_operator(rtxn, index, numbers_db, strings_db, fid, &op)
                     } else {
-                        return Err(fid.as_external_error(FilterError::InternalError))?;
+                        return Ok(RoaringBitmap::new());
```
Comment (Kerollmops) [inline milli/src/search/facet/filter.rs:348]: Do you think this is absolutely valid, though? That when a field can't be found in the `FieldsIdsMap` but has effectively been declared in the `filterableAttributes` it means that no document has ever been declared with this field and therefore we can return an empty set?

File: milli/src/search/facet/filter.rs (hunk 1 above)
Comment (irevoire) [reply milli/src/search/facet/filter.rs:343]: Yes but `Deref` returns the « raw » value and I think here it's better to get the real value in it (with the unescaped quote if there was any)

File: milli/src/search/facet/filter.rs (hunk 2 above)
Comment (irevoire) [reply milli/src/search/facet/filter.rs:61]: Same :joy:

File: milli/src/search/facet/filter.rs (hunk 1 above)
Comment (Kerollmops) [reply milli/src/search/facet/filter.rs:343]: Ho, yeah. Not sure if Deref should return the raw or escaped value here :) But that'll part of another PR 😄

File: milli/src/search/facet/filter.rs (hunk 3 above)
Comment (irevoire) [reply milli/src/search/facet/filter.rs:348]: I think it's valid yes

**Review by Kerollmops — APPROVED** [review]
Thank you for the fix, it looks valid!
bors merge

**Pull request merged**